modules/dists/BlockDist.chpl

```diff
@@ -1407,7 +1422,8 @@ proc BlockDom.dsiGetPrivatizeData() return (dist.pid, whole.dims());
 
 proc BlockDom.dsiPrivatize(privatizeData) {
   var privdist = chpl_getPrivatizedCopy(dist.type, privatizeData(1));
-  var c = new BlockDom(rank=rank, idxType=idxType, stridable=stridable, dist=privdist);
+  var c = new BlockDom(rank=rank, idxType=idxType, stridable=stridable,
+      sparseLayoutType=sparseLayoutType, dist=privdist);
```

Comment:
Strictly speaking, should these read `sparseLayoutType=privdist.sparseLayoutType` in dsiPrivatize()?

Even if it's wrong, I don't think this would cause any issue in practice. `sparseLayoutType` has to be a field in BlockDom simply because it's a generic argument in Block, and BlockDom has a hook to Block. -- Not sure if I am making sense here, though.